app/frontend/src/components/ContributorForm/ContributorForm.tsx

```diff
@@ -0,0 +1,242 @@
+    onChange: (s: string) => void
+  ) => {
+    const currentChoices = value.split(",").filter((v) => !!v);
+    if (currentChoices.includes(option)) {
+      onChange(currentChoices.filter((opt) => opt !== option).join(","));
+    } else {
+      onChange(currentChoices.concat(option).join(","));
+    }
+  };
+
+  return loading ? (
+    <CircularProgress />
+  ) : (
+    <>
+      {success ? (
+        <>
+          <Typography variant="body1">{SUCCESS_MSG}</Typography>
+        </>
+      ) : (
+        <form onSubmit={submit}>
+          <Typography variant="body1">{FILL_IN_THE_FORM}</Typography>
+          <Typography variant="body1">{QUESTIONS_OPTIONAL}</Typography>
+          <TextField
+            inputRef={register}
+            id="ideas"
+            margin="normal"
+            name="ideas"
+            label={IDEAS_LABEL}
+            helperText={IDEAS_HELPER}
+            fullWidth
+            multiline
+            rows={4}
+            rowsMax={6}
+          />
+          <TextField
+            inputRef={register}
+            id="features"
+            margin="normal"
+            name="features"
+            label={FEATURES_LABEL}
+            helperText={FEATURES_HELPER}
+            fullWidth
+            multiline
+            rows={4}
+            rowsMax={6}
+          />
+          <Controller
+            id="contribute"
+            control={control}
+            name="contribute"
+            defaultValue=""
+            render={({ onChange }) => (
+              <FormControl>
+                <FormLabel component="legend">{CONTRIBUTE_LABEL}</FormLabel>
+                <RadioGroup
+                  className={classes.contributeRadio}
+                  aria-label={CONTRIBUTE_ARIA_LABEL}
+                  name="contribute-radio"
+                  onChange={onChange}
+                >
+                  {CONTRIBUTE_OPTIONS.map((opt) => (
+                    <FormControlLabel
+                      key={opt}
+                      value={opt}
+                      control={<Radio />}
+                      label={opt}
+                    />
+                  ))}
+                </RadioGroup>
+              </FormControl>
+            )}
+          />
+          <TextField
+            inputRef={register}
+            id="experience"
+            margin="normal"
+            name="experience"
+            label={EXPERIENCE_LABEL}
+            helperText={EXPERIENCE_HELPER}
+            fullWidth
+            multiline
+            rows={4}
+            rowsMax={6}
+          />
+          <Controller
+            id="contributeWays"
+            control={control}
+            name="contributeWays"
+            defaultValue=""
+            render={({ onChange, value }) => (
+              <FormControl>
+                <FormLabel>{CONTRIBUTE_WAYS_LABEL}</FormLabel>
+                <FormGroup aria-label={CONTRIBUTE_WAYS_ARIA_LABEL}>
+                  {CONTRIBUTE_WAYS_OPTIONS.map(({ name, description }) => (
```

Comment:
I think not destructuring is clearer here - `name` is an important prop for react-hook-form which is a bit confusing
```suggestion
                  {CONTRIBUTE_WAYS_OPTIONS.map((option) => (
```